blockly/src/server/Server.java

```diff
@@ -1233,79 +1233,45 @@ public PositionComponent getHeroPosition() {
         .orElseThrow(() -> MissingComponentException.build(hero, PositionComponent.class));
   }
 
-  /** Throw a fireball upwards. */
-  public void fireballUp() {
+  private void shootFireBall(final Point direction) {
```

Comment:
Bin mir nicht sicher, ob ich hier `Point` als Abstraktion geeignet finde - wir wollen hier ja eine Richtung angeben ("oben", "unten", ..). 

Wäre es nicht geschickter, hier ein Enum mit Richtungen zu verwenden? Dann braucht man nicht beim Aufruf ein künstliches `Point`-Objekt mit Magic Ints zu erzeugen und im if-else wieder auseinanderfriemeln?

Reply:
Möglicherweise könnte das Update der Heldenposition dann auch mit als Hilfsfunktion in das Enum - dann kann man auf der übergebenen Richtung einfach die Hilfsmethode aufrufen (mit der Hero-Position als Argument) und bekommt die gewünschte modifizierte Position zurück.

Reply:
ist ja eigentlich nur ne interne sache hier im server. die up/down/...-methoden haben ja keinen parameter, d.h. das kann super ein kleines internes enum sein (was nicht nach außen exponiert wird).

aus tile würde ich eher nix nehmen, das steckt viel zu tief drin und dient vermutlich auch andere zwecken.

aber letztlich bin ich auch mit dem vorgeschlagenen code zufrieden - du musst für den merge nur diese diskussion schließen :)